libbpf-rs/src/btf/types.rs

```diff
@@ -863,26 +844,333 @@ gen_collection_concrete_type! {
 /// ```
 ///
 /// NonBinding.
+///
 /// ```compile_fail
 ///     BtfKind::Int => {
 ///         /* we don't have access to the variable, but we know the scrutinee is an Int */
 ///     }
 /// ```
+///
+/// Multiple Variants
+/// ```compile_fail
+///     BtfKind::Struct | BtfKind::Union => {
+///         /* we don't have access to the variable, but we know the scrutinee is either a Struct or a Union */
```

Comment:
Nit: Let's not mix in C style comments and stick with `//` unless there is a good reason not to?